_posts/2024-04-10-benchmarks-over-time.md

```diff
@@ -0,0 +1,548 @@
+Python is DuckDB's most popular client, so this is also the most representative of real world performance.
+
+### Export and Replacement Scans
+
+However, we now extend this benchmark in several important ways. 
+In addition to considering raw query performance, we measure import and export performance with several formats: Pandas, Apache Arrow, and Apache Parquet.
+The results of both the join and group by benchmarks are exported to each format.
+
+When exporting to dataframes, we measured the performance in both cases.
+However, when summarizing the total performance, we chose the best performing format at the time.
+This likely mirrors the behavior of performance-sensitive users (as they would likely not write to both formats!).
+In version 0.5.1, DuckDB's performance when writing to and reading from the Apache Arrow format surpassed Pandas. 
+As a result, versions 0.2.7 to 0.4.0 use Pandas, and 0.5.1 onward uses Arrow. 
+
+On the import side, replacement scans allow DuckDB to read those same formats without a prior import step.
+In the replacement scan benchmark, the data that is scanned is the output of the final H20.ai group by benchmark query.
+At the 5GB scale it is a 10 million row dataset.
+Only one column is read, and a single aggregate is calculated. 
+This focuses the benchmark on the speed of scanning the data rather than DuckDB's aggregation algorithms or speed of outputting results.
+The query used follows the format:
+
+```sql
+SELECT 
+    sum(v3) AS v3 
+FROM ⟨dataframe or Parquet file⟩
+```
+
+### Window Functions
+
+We also added an entire series of window function benchmarks. 
+Window functions are a critical workload in real world data analysis scenarios, and can stress test a system in other ways.
+DuckDB has implemented state of the art algorithms to quickly process even the most complex window functions.
+We use the largest table from the join benchmark as the raw data for these new tests to help with comparability to the rest of the benchmark. 
+
+Window function benchmarks are much less common than more traditional joins and aggregations, and we were unable to find a suitable suite off the shelf. 
+These queries were designed to showcase the variety of uses for window functions, but there are certainly more that could be added.
+We are open to your suggestions for queries to add, and hope these queries could prove useful for other systems!
+
+Since the window functions benchmark is new, the window functions from each of the queries included are shown in the appendix at the end of the post.
+
+### Workload Size
+
+We test only the middle 5GB dataset size for the workloads mentioned thus far, primarily because some import and export operations to external formats like Pandas must fit in memory (and we used a MacBook Pro M1 with only 16GB of RAM). 
+Additionally, running the tests for 3 year's of DuckDB versions was time-intensive even at that scale, due to the performance of older versions.
```

Comment:
```suggestion
Additionally, running the tests for X different DuckDB version was time-intensive even at that scale, due to the performance of older versions.
```
I would put an explicit number on how many versions in total.

Reply:
Happy to do that!

Reply:
Made the change, thanks!